extensions/wikia/Recirculation/RecirculationController.class.php

```diff
@@ -58,9 +58,30 @@ public function footer() {
 		$this->response->setVal( 'communityHeaderBackground',
 			$themeSettings->getCommunityHeaderBackgroundUrl() );
 		$this->response->setVal( 'sitename', $wgSitename );
+		$this->response->setVal( 'topWikiArticles', $this->getTopWikiArticles() );
+		// TODO if topWikiArticles is empty show one more N&S article card instead of More from wiki card
 		$this->response->setTemplateEngine( WikiaResponse::TEMPLATE_ENGINE_PHP );
 	}
 
+	private function getTopWikiArticles() {
+		global $wgTitle, $wgCityId;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)